src/personas/data_manager/mod.rs

```diff
@@ -1195,10 +1154,80 @@ impl DataManager {
         }
     }
 
+    fn handle_fragment_refresh(&mut self,
+                               src: XorName,
+                               fragment: FragmentInfo)
+                               -> Result<(), InternalError> {
+        if self.cache
+               .register_needed_fragment_with_another_holder(fragment.clone(), src) {
+            return Ok(());
+        }
+
+        let holders = match self.fragment_refresh_accumulator
```

Comment:
Yes, because this function is used for: 
- refreshing immutable data after node joined/lost.
- refreshing immutable or mutable data after mutation. In case of mutable data, only the modified parts of it are refreshed.